addons/zeus/functions/fnc_ui_toggleFlashlight.sqf

```diff
@@ -0,0 +1,138 @@
+/*
+ * Author: alganthe
+ * Initalises the `Toggle Flashlights` zeus module display
+ *
+ * Arguments:
+ * 0: Flashlight toggle controls group <CONTROL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * onSetFocus = "_this call ace_zeus_fnc_ui_toggleFLashlight"
+ *
+ * Public: No
+*/
+
+#include "script_component.hpp"
+
+disableSerialization;
+
+params ["_control"];
+
+//Generic Init:
+private _display = ctrlparent _control;
+private _ctrlButtonOK = _display displayctrl 1; //IDC_OK
+private _logic = GETMVAR(BIS_fnc_initCuratorAttributes_target,objnull);
+TRACE_1("logicObject",_logic);
+
+_control ctrlRemoveAllEventHandlers "setFocus";
+
+private _unit = effectiveCommander (attachedTo _logic);
+
+// Handles errors
+scopeName "Main";
+private _fnc_errorAndClose = {
+    params ["_msg"];
+    _display closeDisplay 0;
+    deleteVehicle _logic;
+    [_msg] call EFUNC(common,displayTextStructured);
+    breakOut "Main";
+};
+
+if !(isNull _unit) then {
+    switch (false) do {
+        case (_unit isKindOf "CAManBase"): {
+            [LSTRING(OnlyInfantry)] call _fnc_errorAndClose;
+        };
+        case (alive _unit): {
+            [LSTRING(OnlyAlive)] call _fnc_errorAndClose;
+        };
+    };
+};
+
+//Specific on-load stuff:
+private _comboBox = _display displayCtrl 56218;
```

Comment:
Small usability improvement, should set the default to toggle their current state
```
private _enabledDefault = false;
if (isNull _unit) then {
    _enabledDefault = !isFlashlightOn (leader _unit);
};
_comboBox lbSetCurSel [0,1] select _enabledDefault;
```

Reply:
Why `leader _unit`? 
Don't you mean `currentWeapon _unit`?